test/test.js

```diff
@@ -408,3 +409,107 @@ test('MBTiles with two layers', t => {
     t.end();
   }).catch(t.threw);
 });
+
+test('MBTiles with tilestats metadata table returns as expected', t => {
+  Promise.all([
+    geostats(fixturePath('src/tilestats-metadata.mbtiles')),
+    getExpected('tilestats-metadata'),
+  ]).then((output) => {
+    t.deepEqual(sloppySort(output[0]), sloppySort(output[1]), 'expected output');
+    t.end();
+  }).catch(t.threw);
+});
+
+test('[validator] valid stats object', t => {
+  const stats = {
+    layerCount: 1,
+    layers: [
+      {
+        layer: 'test-layer',
+        count: 3,
+        geometry: 'point',
+        attributeCount: 1,
+        attributes: [
+          {
+            attribute: 'test-attribute',
+            count: 3,
+            type: 'number',
+            values: [2, 5, 19],
+            min: 2,
+            max: 19,
+          },
+        ],
+      },
+    ],
+  };
+
+  t.ok(validator(stats));
+  t.end();
+});
+
+test('[validator] invalid stats object - no layers', t => {
+  const stats = {
+    layerCount: 1,
+  };
+
+  const results = validator(stats);
+  t.equal(results[0], 'requires property "layers"', 'expected error message');
+  t.end();
+});
+
+test('[validator] invalid layer object - no layer name', t => {
+  const stats = {
+    layerCount: 1,
+    layers: [
+      {
+        count: 3,
+        geometry: 'point',
+        attributeCount: 1,
+        attributes: [
+          {
+            attribute: 'test-attribute',
+            count: 3,
+            type: 'number',
+            values: [2, 5, 19],
+            min: 2,
+            max: 19,
+          },
+        ],
+      },
+    ],
+  };
+
+  const results = validator(stats);
+  t.equal(results[0], 'requires property "layer"', 'expected error message');
```

Comment:
Would it be worth checking that there is only one error, to ensure unexpected errors didn't happen?